packages/world/src/utils/InventoryUtils.sol

```diff
@@ -340,6 +339,10 @@ library InventoryUtils {
       }
     }
 
+    if (InventoryTypeSlots._length(from, ObjectTypes.Null) > 0) {
```

Comment:
the bug was it wasnt clearing the null slot?

Reply:
yes